### PR TITLE
Task 9: testing

### PR DESCRIPTION
**Pull request by DariaGunina**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,33 +4,52 @@
...
     "@types/react-router-dom": "^5.1.7",
+    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.1",
+    "babel-jest": "^26.6.3",
     "babel-loader": "^8.2.2",
     "clean-webpack-plugin": "^3.0.0",
     "cross-env": "^7.0.3",
     "css-loader": "^5.0.2",
     "css-minimizer-webpack-plugin": "^1.2.0",
+    "enzyme": "^3.11.0",
+    "enzyme-adapter-react-16": "^1.15.6",
```
Comment (mmarkelov) [inline package.json]: Seems like you don't use it

File: src/components/Button/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {Button, ButtonProps} from './Button';
+
+const setupComponent = (render: Function, props: {} = {}) => {
```
Comment (mmarkelov) [inline src/components/Button/Button.test.tsx]: do you need to pass `render` function? As i can see you used `shallow` for both cases

File: src/components/Form/test/Form.test.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import {act, render, screen, waitFor} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import {Form} from '../Form';
+
+test('rendering and submitting a basic Formik form', async () => {
+  const handleSubmit = jest.fn();
+  const handleClose = jest.fn();
+  const props = {
+      onClose: handleClose,
+      onSubmit:handleSubmit,
```
Comment (mmarkelov) [inline src/components/Form/test/Form.test.tsx]: missed space

File: src/components/Form/test/Form.test.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+       userEvent.type(screen.getByLabelText('Overview'), 'Test');
+       userEvent.type(screen.getByLabelText('Runtime'), '60');
+       userEvent.click(screen.getByTestId('genres'));
+       userEvent.click(screen.getByTestId('genre-Horror'));
+   });
+
+   userEvent.click(screen.getByTestId('buttonSubmit'));
+
+   await waitFor(() =>
+       expect(handleSubmit).toHaveBeenCalled()
```
Comment (mmarkelov) [inline src/components/Form/test/Form.test.tsx]: maybe it's better to use `toHaveBeenCalledWith` here?

File: src/components/Form/test/InputField.test.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    const wrapper = render(<InputField {...computedProps} />);
+
+    return {
+        wrapper,
+        props: computedProps,
+    };
+};
+
+describe('InputField,', () => {
```
Comment (mmarkelov) [inline src/components/Form/test/InputField.test.tsx]: Unnecessary comma

File: package.json (hunk 1 above)
Comment (DariaGunina) [reply package.json]: Thanks, this is fixed

File: src/components/Button/Button.test.tsx (hunk 2 above)
Comment (DariaGunina) [reply src/components/Button/Button.test.tsx]: Thanks, this is fixed

File: src/components/Form/test/Form.test.tsx (hunk 3 above)
Comment (DariaGunina) [reply src/components/Form/test/Form.test.tsx]: Thanks, this is fixed

File: src/components/Form/test/Form.test.tsx (hunk 4 above)
Comment (DariaGunina) [reply src/components/Form/test/Form.test.tsx]: Thanks, this is fixed

File: src/components/Form/test/InputField.test.tsx (hunk 5 above)
Comment (DariaGunina) [reply src/components/Form/test/InputField.test.tsx]: Thanks, this is fixed

**Pull request merged**